packages/shared/src/hooks/useActions.ts

```diff
@@ -19,35 +19,40 @@ interface UseActions {
 export const useActions = (): UseActions => {
   const client = useQueryClient();
   const { user } = useAuthContext();
+  const actionsKey = generateQueryKey(RequestKey.Actions, user);
   const { data: actions, isFetched: isActionsFetched } = useQuery(
-    generateQueryKey(RequestKey.Actions, user),
-    getUserActions,
+    actionsKey,
+    async () => {
+      const current = client.getQueryData<Action[]>(actionsKey);
+      const data = await getUserActions();
```

Comment:
Maybe switch these around so we get fresh data after aync fetch because update to `current` could happen while `getUserActions` promise is pending.

Reply:
Adjusted here 5fb4139b29d49c48de7034a0d0550a579dc1a7a0